subworkflows/local/experimental/main.nf

```diff
@@ -5,6 +5,19 @@ include { DIFFERENTIAL }        from '../differential/main.nf'
 include { CORRELATION }         from '../correlation/main.nf'
 include { ENRICHMENT }          from '../enrichment/main.nf'
 
+def postprocess_subworkflow_output( ch_results, ch_tools_args ) {
+    // 1) join results with pathway data
+    // 2) clean up meta data by removing tool arguments and adding pathway name
+    return ch_results
+        .combine(ch_tools_args)
+        .filter{ meta, data, pathway, arg_map -> meta.subMap(arg_map.keySet()) == arg_map }
```

Comment:
if meta has some more elements than 'diff_method' and 'args_diff', would this matching work?

Reply:
Yes, the additional elements would be ignored, because the match is only done on the submap that contains 'diff_method' and 'args_diff' (or 'enr_method' and 'args_enr', etc)